content/rsf-spec/index.md

```diff
@@ -0,0 +1,264 @@
+---
+rfc:
+start_date: 2018-04-04
+pr:
+status: draft
+---
+
+# The Register Serialisation Format
+
+## Summary
+
+This RFC aims to collect in one place the current implementation of RSF so it
+can be added to the specification and it can be evolved with new RFCs when
+required.
+
+The Register Serialisation Format, from now on RSF, is an event log describing
+the evolution of the Register data and metadata.
+
+## Motivation
+
+TODO
+
+## Explanation
+
+### RSF Grammar
+
+RSF is a positional line-based textual format separated by tabs. Each
+line defines a command to apply to a Register state to obtain the next state.
+
+This specification uses the Augmented Backus-Naur Form (ABNF) as defined by
+[RFC5234](https://tools.ietf.org/html/rfc5234) and refined by
+[RFC7405](https://tools.ietf.org/html/rfc7405). It assumes the following
+definitions:
+
+* RFC5234: `ALPHA` (letters), `CRLF` (carriage return, line feed), `DIGIT`
+  (decimal digits), `HEXDIG` (hexadecimal digits) and `HTAB` (horizontal tab).
+* Registers specification: [`CANONREP`][canon-rep] (canonical representation).
+  Note that, in turn, it depends on [RFC8259](https://tools.ietf.org/html/rfc8259).
+
+```abnf
+log              = command *(CRLF command) [CRLF]
+command          = add-item / append-entry / assert-root-hash
+
+assert-root-hash = %s"assert-root-hash" HTAB hash
+
+add-item         = %s"add-item" HTAB CANONREP
+
+append-entry     = %s"append-entry" HTAB type HTAB key HTAB timestamp HTAB hash-list
+type             = "user" / "system"
+key              = alphanum / %x2D / %x5F
```

Comment:
I'm not sure I follow what `alphanum / %x2D / %x5F` means. Can you explain?

Reply:
`alphanum` is defined a few lines below as `alphanum         = ALPHA / DIGIT`, so the `key` definition is: "any ASCII character (A-Z, a-z) or a hypen (%x2D, -) or an underscore (%x5F, _)".

`alpha` and `digit` are defined in https://tools.ietf.org/html/rfc5234#appendix-B.1
`%x2D` and `%x5F` are the hexadecimal codes for hypen and underscore respectively and `/` defines an OR.

Reply:
Ah I missed the definition of alphanum so that makes sense now. 

Does this imply there is only one character in a key? i.e it's either a single ASCII character, hyphen or underscore. It doesn't appear to say that a key can be made up of multiple of these things but I might be reading it wrong still.

Reply:
HA, very good point. I have to add multiple chars :)

I'm wondering if the first char has special treatment in the current implementation? (e.g. can a key start with a hypen: `-foo`? If not, this needs further restriction